### PR TITLE
feat: plot chembench corpus and tiny subset

### PR DESCRIPTION
**Pull request by n0w0f**
## Summary by Sourcery

Add a new plot to the appendix to compare model performance on the ChemBench corpus and the 'tiny' subset, and introduce a script to generate this plot.

New Features:
- Add a new plot to compare model performance on the ChemBench corpus and the 'tiny' subset.

Enhancements:
- Introduce a new script to generate plots comparing model performance on different datasets.

### CODE REVIEW

File: src/tex/appendix.tex [hunk 1]
```diff
@@ -218,6 +218,8 @@ \subsection{Model performance} \label{sec:model_performance_app}
 
 \Cref{fig:performance_per_topic_tiny} shows the same analysis on \chembenchmini.
 
+\Cref{fig:performance_corpus_and_tiny} shows the performance of the models on the \chembench corpus and the \chembenchmini subset. The relative performance difference between both is actually quite similar across most models. This makes \chembenchmini subset a reliable subset for human baseline comparison and particularly valuable for rapid prototyping and initial model assessment phases.
```
Comment (kjappelbaum) [inline src/tex/appendix.tex:221]:
```suggestion
\Cref{fig:performance_corpus_and_tiny} shows the performance of the models on the \chembench corpus and the \chembenchmini subset. The relative performance difference between both is quite similar across most models. This makes \chembenchmini subset a reliable subset for human baseline comparison and particularly valuable for rapid prototyping and initial model assessment phases.
```

I think you also had some nice comments about the preference data - do you want to add them?

File: Snakefile
Comment (kjappelbaum) [inline Snakefile:1]: nice that you also included it here!

**Comment by kjappelbaum** [comment]
very cool to have this!

File: src/tex/appendix.tex (hunk 1 above)
Comment (n0w0f) [reply src/tex/appendix.tex]: added observation about chemical preference on GPT-4 and referenced breakdowns to tables in [da0c066](https://github.com/lamalab-org/chembench-paper/pull/69/commits/da0c0661be4d2e5dc30736e799e0c6503feb2b18)

File: src/tex/appendix.tex [hunk 2]
```diff
@@ -218,6 +220,10 @@ \subsection{Model performance} \label{sec:model_performance_app}
 
 \Cref{fig:performance_per_topic_tiny} shows the same analysis on \chembenchmini.
 
+\Cref{fig:performance_corpus_and_tiny} shows the performance of the models on the \chembench corpus and the \chembenchmini subset. The relative performance difference between both is actually quite similar across most models. 
+This makes \chembenchmini subset a reliable subset for human baseline comparison and particularly valuable for rapid prototyping and initial model assessment phases. 
+An interesting observation is the significant impact of chemical preference tasks on \GPTFour's scores. A detailed breakdown of overall accuracy into scores on different skills and difficulty levels is provided in \Cref{table:performance_table} and \Cref{tab:performance_table_human_subset} for \chembench corpus and the \chembenchmini subset, respectively.
```
Comment (kjappelbaum) [inline src/tex/appendix.tex:225]:
```suggestion
\Cref{fig:performance_corpus_and_tiny} shows the performance of the models on the \chembench corpus and the \chembenchmini subset. The relative performance difference between both is quite similar across most models. 
This makes \chembenchmini subset a reliable subset for human baseline comparison and particularly valuable for rapid prototyping and initial model assessment phases. 

An interesting observation is the significant impact of chemical preference tasks on \GPTFour's scores. A detailed breakdown of overall accuracy into scores on different skills and difficulty levels is provided in \Cref{table:performance_table} and \Cref{tab:performance_table_human_subset} for \chembench corpus and the \chembenchmini subset, respectively.
```

**Pull request merged**